.github/workflows/pb-plugin-tests.yml

```diff
@@ -59,79 +65,164 @@ jobs:
+            vendor
+          key: ${{ runner.os }}-composer-${{ matrix.php }}-${{ hashFiles('**/composer.json', '**/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-composer-${{ matrix.php }}-
+            ${{ runner.os }}-composer-
+
+      - name: Cache WordPress test suite
+        uses: actions/cache@v4
+        with:
+          path: /tmp/wordpress-tests-lib
+          key: ${{ runner.os }}-wordpress-${{ matrix.wordpress }}-${{ hashFiles('bin/install-wp-tests.sh') }}
+
       - name: Install PHP dependencies
         env:
           COMPOSER_AUTH: '{"github-oauth": {"github.com": "${{secrets.PAT_FOR_GITHUB_ACTIONS}}"} }'
         run: |
           export PATH="$HOME/.composer/vendor/bin:$PATH"
-          composer install --no-interaction
+          composer install --no-interaction --prefer-dist
           echo "REPO_NAME=$(basename $(pwd))" >> $GITHUB_ENV
 
       - name: Check Standards
         run: composer standards
 
-      - name: Remove MySQL
-        if: inputs.use_mariadb == true
-        run: sudo apt remove mysql-server-8.0 mysql-common
+      - name: Require Pressbooks
+        if: inputs.requires_pressbooks == true
+        run: |
+          git clone --depth=1 --single-branch https://github.com/pressbooks/pressbooks.git ../pressbooks
+          cd ../pressbooks && composer install --no-dev --prefer-dist
+          cd ../$REPO_NAME
+
+      - name: Require Private Repo
+        if: inputs.requires_private_repo != ''
+        run: |
+          git clone --depth=1 --single-branch https://x-access-token:${{secrets.PAT_FOR_GITHUB_ACTIONS}}@github.com/pressbooks/${{ inputs.requires_private_repo }}.git ../${{ inputs.requires_private_repo }}
+          cd ../${{ inputs.requires_private_repo }} && composer install --no-dev --prefer-dist
+          cd ../$REPO_NAME
+
+      - name: Install WP tests
+        run: bash bin/install-wp-tests.sh wordpress_test root root 127.0.0.1 ${{ matrix.wordpress }}
+
+      - name: Run PHP Tests
+        if: matrix.experimental == true
+        run: composer test
+
+      - name: Run PHP Tests and PCOV
+        if: matrix.experimental == false && inputs.code_coverage == true
+        run: composer test-coverage
+
+      - name: Upload Coverage to Codecov
+        if: matrix.experimental == false && inputs.code_coverage == true
+        run: bash <(curl -s https://codecov.io/bash)
+
+  mariadb-tests:
+    if: inputs.use_mariadb
+    runs-on: ubuntu-20.04
+    services:
+      database:
+        image: mariadb:10.5
+        env:
+          MYSQL_ROOT_PASSWORD: root
+        ports:
+          - 3306:3306
+        options: --health-cmd="mysqladmin ping" --health-interval=5s --health-timeout=5s --health-retries=3
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - php: 8.1
+            wordpress: 6.7
+            experimental: false
+          - php: 8.1
+            wordpress: latest
+            experimental: false
```

Comment:
Probably we want this to be `experimental: true` since we only want a single test running with coverage

Reply:
good catch I added that